tools/noms/staging.py

```diff
@@ -51,13 +53,54 @@ def stage(stagingDir):
             for f in glob.glob(pattern):
                 if os.path.isdir(f):
                     continue
-                fromDir, name = os.path.split(f)
+                from_dir, name = os.path.split(f)
                 if name in exclude:
                     continue
-                toDir = os.path.join(stagingDir, fromDir)
-                if not os.path.exists(toDir):
-                    os.makedirs(toDir)
-                shutil.copy2(f, toDir)
+                to_dir = os.path.join(stagingDir, from_dir)
+                if not os.path.exists(to_dir):
+                    os.makedirs(to_dir)
+                shutil.copy2(f, to_dir)
+    return stage
+
+def HashGlobCopier(index_file, *globs):
+    exclude = ('webpack.config.js',)
+    def stage(stagingDir):
+        rename_dict = dict()
+        for pattern in globs:
+            for f in glob.glob(pattern):
+                if os.path.isdir(f):
+                    continue
+                from_dir, name = os.path.split(f)
+                if name in exclude:
+                    continue
+                to_dir = os.path.join(stagingDir, from_dir)
+                if not os.path.exists(to_dir):
+                    os.makedirs(to_dir)
+
```

Comment:
How about you write a method which yields all paths to consider, then use it here and in GlobCopier? like:

```python
def run_globs(globs, exclude):
  for pattern in globs:
    for f in glob.glob(pattern):
      if os.path.isdir(f):
        continue
      from_dir, name = os.path.split(f)
      if name in exclude:
        continue
      to_dir = os.path.join(staging_dir, from_dir)
        if not os.path.exists(to_dir):
          os.makedirs(to_dir)
      yield (f, to_dir)

def GlobCopier(*globs):
  exclude = ...
  def stage(staging_dir):
    for f, to_dir in run_globs(globs, exclude):
      shutil.copy2(f, to_dir)
  return stage

def HashGlobCopier(*globs):
  exclude = ...
  def stage(staging_dir):
    for f, to_dir in run_globs(globs, exclude):
      ...
  return stage
```